open-craft/edx-platform
# Set SWIFT_CONTENT_LENGTH_FROM_FD=False

Works around a bug with Django SWIFT storage uploads in edx-platform encountered since upgrading to `django-swift-storage==1.2.18` (cf https://github.com/edx/edx-platform/pull/17487).

Also adds `SWIFT_LAZY_CONNECT=True` to `cms.envs.openstack`, to balance with the lms settings.

**JIRA tickets**: OC-4133

**Dependencies**:  https://github.com/dennisv/django-storage-swift/pull/100

**Sandbox URL**:

* LMS: https://oc4133.sandbox.opencraft.hosting/
* Studio: https://studio-oc4133.sandbox.opencraft.hosting/

**Merge deadline**: ASAP

**Testing instructions**:

Use Appserver 1 to demonstrates that the following actions fail, and Appserver 2 to demonstrate that they succeed.

1. In Studio, Export the DemoX course (or any other large course).
1. In Studio, Import the DemoX course (or any other large course) into a new course.
1. In LMS, using the Instructor > Data Download area, download any student reports, e.g. grade reports or student profile CSVs.  The downloaded files should be non-empty.
1. In LMS Discussion area, post a large (~1Mb) image.  This should work, and the image should display in the post.

**Author notes and concerns**:

1. Once https://github.com/dennisv/django-storage-swift/pull/100 merges, will need to update the requirements to stop using this branch and use the newly tagged release version instead.

**Reviewers**
